test/functional/ui/messages_spec.lua

```diff
@@ -1096,6 +1097,29 @@ vimComment     xxx match /\s"[^\-:.%#=*].*$/ms=s+1,lc=1  excludenl contains=@vim
       {4:Press ENTER or type command to continue}^                     |
     ]]}
   end)
+  it('msgfunc disables messages output', function()
+    screen:try_resize(40,6)
+    source([[
+    function! Func(method, kind, chunks, overwrite) abort
+      let g:method = a:method
+      let g:kind = a:kind
+      let g:chunks = a:chunks
+    endfunction
+    set msgfunc=Func
+    ]])
+    feed(':echo "hello"<cr>')
```

Comment:
Try putting another `screen:expect()` right before this `feed` I believe the same one you have below could work **to solve the warning** you posted as a comment. 👍

Reply:
```
warning: Screen test succeeded immediately. Try to avoid this unless the
purpose of the test really requires it.
There are intermediate states between the two identical expects.
Use screen:snapshot_util() or screen:redraw_debug() to find them, and add them
to the test if they make sense.
```

Well it is good?